tests/unit/mathematics/linalg/MeanEigen3_unittest.cc

```diff
@@ -82,5 +82,180 @@ TEST(MeanEigen3, Eigen3_dynamic_explicit_eigen3_backend_int)
 	EXPECT_NEAR(linalg::mean<linalg::Backend::EIGEN3>(a), 4.5, 1E-15);
 }
 
+TEST(MeanEigen3, SGMatrix_asymmetric_eigen3_backend_with_diag_float)
+{
+	const index_t m = 2;
+	const index_t n = 3;
+	SGMatrix<float64_t> mat(m, n);
+
+	for (index_t i = 0; i < m; ++i)
+	{
```

Comment:
btw it would be nicer if one could "read off" the matrix in such unit tests -- so that I can directly check what is going on. This loop is hard to decipher.